### PR TITLE
Brokerage Models Order Supported Types Check

### LINKED ISSUE #6937: Brokerage Models Support More Order Types than the Brokerage Implementation
#### Expected Behavior

The brokerage models only let us create orders that the respective brokerage integration supports.

#### Actual Behavior

- The [ZerodhaBrokerageModel](https://github.com/QuantConnect/Lean/blob/master/Common/Brokerages/ZerodhaBrokerageModel.cs) lets us create orders of any order type, but the [ZerodhaBrokerage](https://github.com/QuantConnect/Lean.Brokerages.Zerodha/blob/master/QuantConnect.ZerodhaBrokerage/ZerodhaBrokerage.cs#L514:L531) only supports a subset of the order types.
- The [TradierBrokerageModel](https://github.com/QuantConnect/Lean/blob/master/Common/Brokerages/TradierBrokerageModel.cs) lets us create orders of any order type except MOO and MOC, but the [TradierBrokerage](https://github.com/QuantConnect/Lean.Brokerages.Tradier/blob/master/QuantConnect.TradierBrokerage/TradierBrokerage.cs#L1450:L1469) only supports market, limit, stop limit, and stop market orders.

There may be other brokerages with this issue

#### Potential Solution

- Update the `CanSubmitOrder` of each brokerage model to reflect the supported order types of the brokerage integration.

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [ ] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by abhi-kr-2100**
Modifies `CanSubmitOrder` to validate order types.

#### Description
Some brokerage models don't check the order type in `CanSubmitOrder`. As a result, unimplemented order types don't cause `CanSubmitOrder` to return false as it should.

This pull request adds order validation to the following brokerage models:
* `FTXBrokerageModel.cs`
* `InteractiveBrokersBrokerageModel.cs`
* `SamcoBrokerageModel.cs`
* `TradierBrokerageModel.cs`
* `ZerodhaBrokerageModel.cs`

Other brokerage model have not been modified if:
* they already had order validation
* their implementation was not found

#### Related Issue
Closes #6937 

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. (Not applicable because the logic is very simple; tests would simply be repeating the code.) 
- [ ] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Brokerages/FTXBrokerageModel.cs [hunk 1]
```diff
@@ -27,6 +28,14 @@ namespace QuantConnect.Brokerages
     /// </summary>
     public class FTXBrokerageModel : DefaultBrokerageModel
     {
+        private readonly OrderType[] _supportedOrderTypes =
```
Comment (Martin-Molinero) [inline Common/Brokerages/FTXBrokerageModel.cs:31]: > OrderType[]

Let's replace this for a `HashSet<OrderType>` 👍

**Review by Martin-Molinero — COMMENTED** [review]
Hi @abhi-kr-2100! Thanks for the contribution PR looks good, just leaving 1 request

File: Common/Brokerages/FTXBrokerageModel.cs (hunk 1 above)
Comment (abhi-kr-2100) [reply Common/Brokerages/FTXBrokerageModel.cs]: Thank you for the review. I've made the requested change.

**Review by Martin-Molinero — APPROVED** [review]
Thanks @abhi-kr-2100! Push a small commit with further normalization and improved error messaging

**Pull request merged**